lxd/db/generate/db/stmt.go

```diff
@@ -87,6 +87,10 @@ func (s *Stmt) GenerateSignature(buf *file.Buffer) error {
 }
 
 func (s *Stmt) objects(buf *file.Buffer) error {
+	if strings.HasPrefix(s.kind, "objects-by") {
```

Comment:
Can you add a comment here explaining what its doing please?

Reply:
Sorry, the change there is basically what the first half of PR description is. I've added a commit message to describe it better.